### PR TITLE
chore(input format): more tests and diagnostic info.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

notes：

- parquet  can not load all types：https://github.com/datafuselabs/databend/issues/8661
- help to debug
    -  https://github.com/datafuselabs/databend/issues/8643
    - https://github.com/datafuselabs/databend/issues/8579

Closes #issue

### CODE REVIEW

File: src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs [hunk 1]
```diff
@@ -58,8 +60,17 @@ impl InputFormatCSV {
                     let err_msg = format_column_error(schema, c, col_data, &e.message());
                     return Err(csv_error(&err_msg, path, row_index));
                 };
-                if reader.must_eof().is_err() {
-                    let err_msg = format_column_error(schema, c, col_data, "bad field end");
+                let mut next = [0u8; 1];
+                let readn = reader.read(&mut next[..])?;
+                if readn > 0 {
+                    let remaining = col_data.len() - reader.pos + 1;
+                    let err_msg = format!(
```
Comment (Xuanwo) [inline src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs:67]: Nice!

File: src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs [hunk 2]
```diff
@@ -58,8 +60,17 @@ impl InputFormatCSV {
...
+                        "bad field end: next={}, len={}.",
```
Comment (Xuanwo) [inline src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs:68]: At first glance, I can't understand what `next` and `len` mean. How about providing more context?

The trailing `.` can be removed.

File: tests/suites/1_stateful/05_formats/05_04_tsv/05_04_01_tsv_multi_file.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    c4 int,
+    c5 int,
+    c6 int,
+    c7 int,
+    c8 int,
+    c9 int
+);" | $MYSQL_CLIENT_CONNECT
+
+# do copy
+echo "SET GLOBAL max_threads = 32;" | $MYSQL_CLIENT_CONNECT
```
Comment (Xuanwo) [inline tests/suites/1_stateful/05_formats/05_04_tsv/05_04_01_tsv_multi_file.sh:34]: Will this affect other tests?

Comment (youngsofun) [reply tests/suites/1_stateful/05_formats/05_04_tsv/05_04_01_tsv_multi_file.sh:34]: yes, I forgot to set it back, thank you!

File: src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs (hunk 2 above)
Comment (youngsofun) [reply src/query/pipeline/sources/src/processors/sources/input_formats/impls/input_format_csv.rs]: refined

**Review by Xuanwo — APPROVED** [review]
Great PR!

**Review by BohuTANG — APPROVED** [review]

**Comment by youngsofun** [comment]
q5 failed twice, retrying again.

**Comment by BohuTANG** [comment]
> q5 failed twice, retrying again.

Not related to this PR, issued https://github.com/datafuselabs/databend/issues/7986

**Pull request merged**